analyzers/src/SonarAnalyzer.CSharp/Rules/CollectionEmptinessChecking.cs

```diff
@@ -138,11 +89,61 @@ private static bool TryGetCountCall(ExpressionSyntax expression, SemanticModel s
             return true;
         }
 
-        private static bool IsMethodCountExtension(IMethodSymbol methodSymbol)
+        private static bool IsMethodCountExtension(IMethodSymbol methodSymbol) =>
+            methodSymbol.Name == "Count" &&
+            methodSymbol.IsExtensionMethod &&
+            methodSymbol.ReceiverType != null;
+
+        internal readonly struct CountType
```

Comment:
Can you please reorder members in this struct according to https://github.com/SonarSource/sonar-dotnet/blob/master/docs/coding-style.md ? 

There should be fields, empty line, properties, empty line, constructor, other methods (separated by empty line)